### DISCUSSION TITLE
How to request `accessToken` (via authorization code)?

### DISCUSSION THREAD

**Discussion by dccarmo** [neet/masto.js#1055] [Q&A] [answered]
Hi, 

First of all, thank you for the masto.js, it's such a great library to work with. It's so elegantly built, I really appreciate the effort.

I wanted to understand if we're missing something, but we couldn't find a way to request an `accessToken` from the library, is that supported? 

More specifically, request `accessToken` using the OAuth API:

- first calling `GET /oauth/authorize`
- then calling `POST /oauth/token`

Right now we're doing this flow using Axios and then, once we have the `accessToken`, we start using masto.js for all other calls.

Are we correct to assume that masto.js doesn't support this initial flow?

I found out on a [comment](https://github.com/neet/masto.js/discussions/1010#discussioncomment-7549334) here that OAuth support is pretty new to masto.js, so maybe we're just missing implementing those calls? I'd be happy to contribute with a PR.

**Comment by neet** [comment]
@dccarmo Hi, thank you for posting a question. Although It's not well documented at this time, you can find a working code at tests directory:

https://github.com/neet/masto.js/blob/fce7a60e16d59241ab5f11f65ec095f70fb9f6e6/tests/oauth/token.spec.ts#L5-L16

Basically, you need to initialise OAuth API dedicated client via `createOAuthAPIClient` and then everything works almost the same as REST API.

We have room for documentation so I'm planning to improve it in the future 🙏

**Comment by dccarmo** [comment]
@neet thank you for your response! Does `oauth.token.create` only support the `grant_type: "password"`? Because for the `grant_type = "authorization_code"`, we would need to request the authorization code first (by calling `GET /oauth/authorize`).

**Reply by neet** [reply]
I didn't read the post title 😅

As you pointed out, it's currently providing a type definitions for only `grant_type: password`. I'll test if I can support `authorization_code` as well.

**Reply by neet** [reply]
I made an issue #1057

**Reply by neet** [reply, marked as answer]
I shipped `grant_type: authorization_code` and `client_credentials` in v6.7.0.
https://github.com/neet/masto.js/releases/tag/v6.7.0

Try updating masto.js to the latest version and verify it works. Let me know if you still have a trouble!

**Reply by dccarmo** [reply]
That worked perfectly! Thank you so much!
